### ISSUE TITLE
`needless_continue` false positive when match all arm is not continue

### ISSUE THREAD

**Issue by ericswpark** [rust-lang/rust-clippy#14718]
### Summary

When a match is done within a loop block like the following, clippy emits a `needless_continue` lint:

```rust
loop {
  match some_func().unwrap_err() {
    SomeBenignError => {
      continue;
    }
    SeriousError => {
      // Send diagnostics to server and panic!
      // ...
    }
    err => {
      error!("Error in internal library: {err}");
    }
  }
}
```

`cargo clippy` suggests I can omit the `SomeBenignError` match arm, but the behavior in the named match-all arm `err` is not continue and can have possible side effects depending on what statements are in there.

### Lint Name

needless_continue

### Reproducer

I tried this code:

```rust
loop {
  match some_func().unwrap_err() {
    SomeBenignError => {
      continue;
    }
    SeriousError => {
      // Send diagnostics to server and panic!
      // ...
    }
    err => {
      error!("Error in internal library: {err}");
    }
  }
}
```

I saw this happen:

```
warning: this `continue` expression is redundant
  --> src\main.rs:76:21
   |
76 |                     continue;
   |                     ^^^^^^^^
   |
   = help: consider dropping the `continue` expression
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_continue
   = note: `-W clippy::needless-continue` implied by `-W clippy::pedantic`
   = help: to override `-W clippy::pedantic` add `#[allow(clippy::needless_continue)]`
```

I expected to see this happen: no linting warning if the match-all arm is not continue

### Version

```text
rustc 1.86.0 (05f9846f8 2025-03-31)
binary: rustc
commit-hash: 05f9846f893b09a1be1fc8560e33fc3c815cfecb
commit-date: 2025-03-31
host: x86_64-pc-windows-msvc
release: 1.86.0
LLVM version: 19.1.7
```

**Comment by profetia** [comment]
I think clippy means that you can drop the `continue` statement, not the match arm.

**Comment by Jarcho** [comment]
Yes. The suggestion is telling you to remove the continue statement, not the entire match arm.

**Comment by ericswpark** [comment]
I see, sorry for misunderstanding the output!
